working-with-openai/open-ai-api-endpoints.md

```diff
@@ -0,0 +1,156 @@
+# Using the OpenAI API
+
+## Goals
+
+Now that we know a bit about the OpenAI models, lets learn about the OpenAI endpoints. 
+
+Our goals for this lesson are to:
+- Recognize different OpenAI endpoints and how they are used
+- Write our first OpenAI request in Postman
+
+## Exploring OpenAI Endpoints
+
+In the last lesson, we looked at the different models associated with OpenAI. It may come as no surprise that each model maps to a specific endpoint we can use to access different AI functionalities. Below is a non-exhaustive list of OpenAI's endpoints and the models they map to:
+
+| Endpoint | Verb | Model(s) | Usage |
+| -------- | ---- | -------- | ----- |
+| https://api.openai.com/v1/chat/completions | POST | GPT 3.5 Turbo/GPT 4.0/ GPT 4.0 Turbo | Generating a response to a set of messages |
+| https://api.openai.com/v1/images/generations | POST | DALL-E | Generate a new image from a given prompt |
+| https://api.openai.com/v1/audio/speech | POST | TTS | Generate audio from input |
+| https://api.openai.com/v1/audio/transcriptions | POST | Whisper | Generate a transcription of audio input |
+| https://api.openai.com/v1/embeddings | POST | Embeddings | Creates a numerical  representation of text |
+| https://api.openai.com/v1/moderations | POST | Moderations | Checks to see if text could be potentially harmful | 
+| https://api.openai.com/v1/models | GET | None | Returns a list of OpenAI models | 
+
+We will focus on using the chat completions endpoint for now, but  each endpoint can be called similarly. Each request body will require some form of input at a minimum and most will require the specific model you would like to use as well. The response you receive will also vary from request to request so make sure you have an understanding of the shape of a response before jumping right in. The full list of OpenAI endpoints along with examples of a request and response body can be found [here](https://platform.openai.com/docs/api-reference/introduction). OpenAI also provides a [model compatability chart](https://platform.openai.com/docs/models/model-endpoint-compatibility) that allows you to see which models work with each endpoint.
+
+## The Chat Completions Endpoint
+
+Before we jump right into integrating the API into a project, let's explore the chat completions endpoint. This particular endpoint is used to mimic and then continue a conversation between the user and an artificial intelligence. In terms of chatGPT, the chat completion endpoint is what is used to interact with the GPT models. The term "chat completion" simply refers to the fact that the response will be the next logical reply to the conversation that is going on. 
+
+### Chat Completion Request Body
+
+As mentioned earlier, the chat completion endpoint is: 
+```
+    POST https://api.openai.com/v1/chat/completions 
+``` 
```

Comment:
I was thinking another way to structure this to highlight the info could be:

> As mentioned earlier, the chat completion endpoint takes a `POST` request at the URL: 
> ```
> https://api.openai.com/v1/chat/completions 
> ``` 

I want to remove the indentation at least since it isn't something we've done in other lessons and might look a little out of place.

(I'm sorry I couldn't add it as a suggestion, GitHub suggestions don't play nice with code blocks since they use the same end demarcation 😭)

Reply:
Done!